crates/bin/c8y-device-management/src/main.rs

```diff
@@ -68,6 +69,11 @@ async fn main() -> anyhow::Result<()> {
     log_actor.with_c8y_http_proxy(&mut c8y_http_proxy_actor)?;
     log_actor.with_mqtt_connection(&mut mqtt_actor)?;
 
+    //Instantiate health monitor actor
+    let health_actor = HealthMonitorBuilder::new(PLUGIN_NAME);
+    mqtt_actor.mqtt_config = health_actor.set_init_and_last_will(mqtt_actor.mqtt_config);
```

Comment:
Although there's nothing wrong with this approach, I wish this init and last-will messages were also exchanged with the MQTT actor via the `Config` object of the `ServiceConsumer` to make it similar to how the `TopicFilters` are exchanged. But, since the init and last-will messages can only be registered once with the MQTT actor, this is probably fine.

Reply:
Good point. It would be indeed nice to have this done when the health actor is connected to the mqtt actor. Regarding the issue that at most one last will message can be set, one can simply raise a `LinkError` if two actors attempt to set a last-will. This can be done as a follow-up task.